### PR TITLE
Implement ROW_GROUP_SIZE_BYTES for Parquet writer

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR implements the `ROW_GROUP_SIZE_BYTES` parameter for the Parquet writer, similar to Iceberg's [`row-group-size-bytes`](https://iceberg.apache.org/docs/latest/configuration/).

This parameter limits the size of the buffered data per row group. While we already have a default limit of `ROW_GROUP_SIZE` of 122880, this offers a different way to limit the row group size that can adapt to when there are many columns / large strings.

```sql
copy (
    select range c0,
           range c1,
           range c2,
           range c3,
           range c4,
           range c5,
           range c6,
           range c7,
    from range(50000)
) to '__TEST_DIR__/tbl.parquet' (ROW_GROUP_SIZE_BYTES '1mb');
-- (ROW_GROUP_SIZE_BYTES 1000000) would be equivalent
select max(row_group_num_rows) from parquet_metadata('tbl.parquet');
-- 16384
```

I've currently made the default setting of `ROW_GROUP_SIZE_BYTES` unlimited, but we could consider following Iceberg's default of 128MB. This would create a more pleasant experience for users, as they would not run into unexpected scenarios where writing 20 columns is fine, but writing 200 columns may give them memory issues.

Happy to receive feedback, and please let me know what you think about the default setting!

### CODE REVIEW

File: extension/parquet/parquet_extension.cpp [hunk 1]
```diff
@@ -116,6 +116,7 @@ struct ParquetWriteBindData : public TableFunctionData {
 	vector<string> column_names;
 	duckdb_parquet::format::CompressionCodec::type codec = duckdb_parquet::format::CompressionCodec::SNAPPY;
 	idx_t row_group_size = RowGroup::ROW_GROUP_SIZE;
+	idx_t row_group_size_bytes = 134217728; // 128MB (1 << 27)
```
Comment (Mytherin) [inline extension/parquet/parquet_extension.cpp:119]: Maybe it makes sense to default this to e.g. 1KB per row? That way the default scales with the user-specified `ROW_GROUP_SIZE`. For our default row group size that would be 128MB. But that would mean that if the user increases the row group size manually (to e.g. 1 million rows) the `row_group_size_bytes` would automatically scale with the user-specified row group size.

Comment (lnkuiper) [reply extension/parquet/parquet_extension.cpp:119]: So just to clarify, if a user sets only the `ROW_GROUP_SIZE` to N, then we flush the row group when we either exceed N rows, or N kb.

If a user sets both `ROW_GROUP_SIZE` to N and `ROW_GROUP_SIZE_BYTES` to M, we flush the row group when we exceed N rows or M bytes.

If a user sets only `ROW_GROUP_SIZE_BYTES` to M, we flush the row group when we exceed the default 122880 rows, or M bytes.

So `ROW_GROUP_SIZE` has an effect on `ROW_GROUP_SIZE_BYTES`, but not the other way around.

Is that what you mean? That sounds reasonable

Comment (Mytherin) [reply extension/parquet/parquet_extension.cpp:119]: Yes, exactly, which in the current implementation means that when `ROW_GROUP_SIZE_BYTES` is not set it defaults to `ROW_GROUP_SIZE * 1KB`.

**Comment by lnkuiper** [comment]
I think this is ready to go!

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**